src/themes/default/components/core/ProductGallery.vue

```diff
@@ -63,34 +38,44 @@
 
 <script>
 import { ProductGallery } from '@vue-storefront/core/modules/catalog/components/ProductGallery.ts'
-import ProductGalleryZoom from './ProductGalleryZoom'
+import ProductGalleryOverlay from './ProductGalleryOverlay'
 import NoSSR from 'vue-no-ssr'
 import VueOfflineMixin from 'vue-offline/mixin'
+const ProductGalleryCarousel = () => import(/* webpackChunkName: "vsf-product-gallery-carousel" */ './ProductGalleryCarousel.vue')
 
 export default {
   components: {
+    ProductGalleryCarousel,
     'no-ssr': NoSSR,
-    ProductGalleryZoom
+    ProductGalleryOverlay
   },
   mixins: [ProductGallery, VueOfflineMixin],
   watch: {
     '$route': 'validateRoute'
   },
   data () {
     return {
-      loaded: true,
-      carouselTransitionSpeed: 0
+      isZoomOpen: false,
```

Comment:
oh here we have isZoomOpen, so i think handle escape event should be here also

Reply:
yeah indeed, i moved it to this component.
But this makes me wonder about the right to exist of ProductGallery.ts. Because so much of the logic is now in the theme file, why still use the minimal ProductGallery.ts in core?

Reply:
generally it's a theme case, so yeah, maybe in future we'll get rid of it; but not now yet, it still could contain some utilities needed in every theme